### PR TITLE
6724: Clone remote GitHub repo with SSH

### LINKED ISSUE #6724: Custom Upstream doc: can no longer `git push` to repo cloned via HTTPS
In the "Create a Custom Upstream" document, ["Create and Host the Repository Remotely" section](https://pantheon.io/docs/create-custom-upstream#create-and-host-the-repository-remotely), the instructions are to clone the repository using the HTTPS URL.  Later, in the "Pull in Core from Pantheon's Upstream" section, the instructions say to `git push origin master`, but Github stopped allowing password authentication in Aug 2021.

## Expected behavior
* User is able to `git push origin master` when following the directions exactly as they are written

## Actual behavior
* Github rejects the push unless user sets up authentication using a personal access token.  Screenshot:
github-auth-requirements

### Steps to reproduce the problem

1. Clone a git repo using the HTTPS URL, without setting up personal access token authentication.  Commit a change, then `git push origin master`

## Suggested Resolution
* Revise instructions to include setting up ssh authentication or personal access token authentication.  I prefer ssh and it seems simpler to me because I'm used to doing it.

## Additional information

Github:
```
Support for password authentication was removed on August 13, 2021. Please use a personal access token instead.
Please see https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/ for more information.
```
[Link to URL above](https://github.blog/2020-12-15-token-authentication-requirements-for-git-operations/)

### PR DESCRIPTION
**Pull request by EdwardAngert**
Closes #6724 

## Summary

**[Create a Custom Upstream](https://pantheon.io/docs/create-custom-upstream)** - Update the GitHub remote repo instructions to use SSH.

## Effect

The following changes are already committed:

*
*

## Remaining Work and Prerequisites

The following changes still need to be completed:

- [ ] List any outstanding work here

### Dependencies and Timing

- [ ] Other prerequisites that must be completed before merging this PR

**Release**:
- [x] When ready
- [ ] After date: $DATE

--------------------------------------------------

## Post Launch

**Do not remove** - To be completed by the docs team upon merge:

- [ ] Redirect `/docs/old-path/` => `/docs/new-path/` (if applicable)
- [ ] Include/exclude pages ^ respectively within docs search service provider (if applicable)
- [ ] For Heroes - add a props post to the [discussion board](https://discuss.pantheon.io/c/pantheon-platform/documentation/17).
- [ ] Remove from the [project board](https://github.com/pantheon-systems/documentation/projects/14)

### CODE REVIEW

**Comment by pantheondocs** [comment]
Lighthouse Audit Results:

| Description | [Live](https://pantheon.io/docs) | [pr-6856](https://pr-6856--pantheon-docs.my.pantheonfrontend.website/docs) | Difference |
| --- | --- | --- | --- |
| Performance Score | `57.99999999999999` | `54` | `-4` |
| Best Practice Score | `79` | `79` | `0` |
| SEO Score | `96` | `96` | `0` |
| Accessibility Score | `96` | `96` | `0` |

Lighthouse Audit Reports:

- [`pr-6856` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/31369/artifacts/0//lighthouse_data/lighthouse-audit-dev.html)
- [`Pantheon Docs` HTML Report](https://circleci.com/gh/pantheon-systems/documentation/31369/artifacts/0//lighthouse_data/lighthouse-audit-reference.html)


[Preview Environment](https://pr-6856--pantheon-docs.my.pantheonfrontend.website/docs)

File: source/content/create-custom-upstream.md [hunk 1]
```diff
@@ -49,20 +51,20 @@ Choose your preferred Git host:
 
 1. Click **Create Repository**.
 
-1. Copy the repository URL (HTTPS), found in the Quick setup section:
+1. Copy the repository URI (SSH), found in the **Quick setup** section:
```
Comment (EdwardAngert) [inline source/content/create-custom-upstream.md:54]: I've called it a URI here, but I'm not sure that it's technically correct

**Comment by EdwardAngert** [comment]
[preview](https://pr-6856--pantheon-docs.my.pantheonfrontend.website/docs/create-custom-upstream#create-and-host-the-repository-remotely)

File: source/content/create-custom-upstream.md (hunk 1 above)
Comment (namespacebrian) [reply source/content/create-custom-upstream.md:54]: I googled this and the [first result was by Daniel Miessler](https://danielmiessler.com/study/difference-between-uri-url/) who is someone I already greatly respect.  

The TL;DR: 
- You're correct to use URI
- "All URLs are URIs, but not all URIs are URLs." 
- If it implies or contains a protocol component (http:// or equiv) then it's a URL.  Looks like github's ssh reference does _not_ contain an explicit protocol indication

**Review by namespacebrian — COMMENTED** [review]
I have reservations about the change to `source/images/github-avoid-readme.png`..

I believe this image is used in instructions for creating a github repo where you want to import an existing git history--e.g. when creating a new custom upstream.  In those circumstances it's important not to intitialize the repo with a README (or 'initialize' it at all) because that starts a new git history. 

It looks like the change removes the big red box indicating it should not be initialized.  I see it also reflects that the github UI has changed to add more repo-initialization options.  I would update the image to have a (possibly less-heavy-handed?) red X over the entire "initialize" section.  I think the 'avoid' in the image filename makes it clear that the point of this image is to tell people not to select those options.

Incidentally, on this subject, when the "create custom upstream" documentation was originally written, BitBucket did not have similar "initialize repo" options, but now it does.  About a month ago I helped someone who was following that documentation and got stuck because they used BitBucket and initialized the repo with things because "it seemed like a good idea at the time" and that path in the documentation didn't loudly tell them it's important not to do that.

**Comment by namespacebrian** [comment]
[Old image](https://github.com/pantheon-systems/documentation/blob/main/source/images/github-avoid-readme.png)
[Changed version without red X](https://github.com/pantheon-systems/documentation/blob/56cff59c7111dd57e617dbfb68bbafe7e0cb4ff4/source/images/github-avoid-readme.png)

**Review by namespacebrian — APPROVED** [review]
Well, looking at the preview, I see that the text instructions do say not to select those options and the screenshot depicts them as unselected... and I also see that the BitBucket instructions also say to unselect initialization options.  I'm approving as-is, with my comments simply for your consideration.

**Review by michellecolon-pantheon — APPROVED** [review]
LGTM

**Pull request merged**